src_c/_sdl2/video.pyx

```diff
@@ -0,0 +1,415 @@
+    int pgSurface_Check(object surf)
+    SDL_Surface* pgSurface_AsSurface(object surf)
+    void import_pygame_surface()
+    SDL_Rect *pgRect_FromObject(object obj, SDL_Rect *temp)
+    void import_pygame_rect()
+
+import_pygame_surface()
+import_pygame_rect()
+
+class RendererDriverInfo:
+    def __repr__(self):
+        return "<%s(name: %s, flags: 0x%02x, num_texture_formats: %d, max_texture_width: %d, max_texture_height: %d)>" % (
+            self.__class__.__name__,
+            self.name,
+            self.flags,
+            self.num_texture_formats,
+            self.max_texture_width,
+            self.max_texture_height,
+        )
+
+def get_drivers():
+    cdef int num = SDL_GetNumRenderDrivers()
+    cdef SDL_RendererInfo info
+    cdef int ind
+    for ind from 0 <= ind < num:
+        SDL_GetRenderDriverInfo(ind, &info)
+        ret = RendererDriverInfo()
+        ret.name = info.name
+        ret.flags = info.flags
+        ret.num_texture_formats = info.num_texture_formats
+        ret.max_texture_width = info.max_texture_width
+        ret.max_texture_height = info.max_texture_height
+        yield ret
+
+def get_grabbed_window():
+    """return the Window with input grab enabled,
+       or None if input isn't grabbed."""
+    cdef SDL_Window *win = SDL_GetGrabbedWindow()
+    cdef void *ptr
+    if win:
+        ptr = SDL_GetWindowData(win, "pg_window")
+        if not ptr:
+            return None
+        return <object>ptr
+    return None
+
+cdef class Window:
+    DEFAULT_SIZE = 640, 480
+
+    _kwarg_to_flag = {
+        'opengl': _SDL_WINDOW_OPENGL,
+        'vulkan': _SDL_WINDOW_VULKAN,
+        'hidden': _SDL_WINDOW_HIDDEN,
+        'borderless': _SDL_WINDOW_BORDERLESS,
+        'resizable': _SDL_WINDOW_RESIZABLE,
+        'minimized': _SDL_WINDOW_MINIMIZED,
+        'maximized': _SDL_WINDOW_MAXIMIZED,
+        'input_grabbed': _SDL_WINDOW_INPUT_GRABBED,
+        'input_focus': _SDL_WINDOW_INPUT_FOCUS,
+        'mouse_focus': _SDL_WINDOW_MOUSE_FOCUS,
+        'allow_highdpi': _SDL_WINDOW_ALLOW_HIGHDPI,
+        'foreign': _SDL_WINDOW_FOREIGN,
+        'mouse_capture': _SDL_WINDOW_MOUSE_CAPTURE,
+        'always_on_top': _SDL_WINDOW_ALWAYS_ON_TOP,
+        'skip_taskbar': _SDL_WINDOW_SKIP_TASKBAR,
+        'utility': _SDL_WINDOW_UTILITY,
+        'tooltip': _SDL_WINDOW_TOOLTIP,
+        'popup_menu': _SDL_WINDOW_POPUP_MENU,
+    }
+
+    def __init__(self, title='pygame',
+                 size=DEFAULT_SIZE,
+                 position=WINDOWPOS_UNDEFINED,
+                 int fullscreen=0, **kwargs):
+        """ Create a window with the specified position, dimensions, and flags.
+
+        :param title str: the title of the window, in UTF-8 encoding
+        :param size tuple: the size of the window, in screen coordinates (width, height)
+        :param position: a tuple specifying the window position, WINDOWPOS_CENTERED, or WINDOWPOS_UNDEFINED.
+        :param fullscreen int: 0: windowed mode
```

Comment:
Instead of Window(fullscreen=2), how about Window(fullscreen_desktop=True) ?
Window(fullscreen=1) and Window(fullscreen=1) would still exist.

Reply:
It seems a bit confusing to have separate parameters for mutually exclusive options. Maybe fullscreen=pg.FULLSCREEN, fullscreen=pg.FULLSCREEN_DESKTOP, fullscreen=0?